src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -449,10 +449,15 @@ Copyright (c) .NET Foundation. All rights reserved.
                                     Value="$(TieredCompilationQuickJitForLoops)" />
 
     <RuntimeHostConfigurationOption Include="System.StartupHookProvider.IsSupported"
-                                    Condition="'$(StartupHookSupport)' != ''"
+                                    Condition="'$(StartupHookSupport)' != '' And '$(PublishTrimmed)' == 'true'"
```

Comment:
This is going to cause a problem because setting `StartupHookSupport` or `CustomResourceTypesSupport` properties even without trimming should still produce the same behavior in the app. So we still need to set the runtimeconfig.json setting even if we aren't trimming.

Reply:
Fixed. Chatted with @pranavkm offline and I'm moving the PublishTrimmed setting up to the props instead of targets for wasm.